### PR TITLE
Pullquote styling

### PR DESCRIPTION
**Pull request by joshdarby**
## Changes

This pull request makes the following changes:

- Adds pullquote styling based on the mockups
- Adds support for theme colors in the block editor color picker

Desktop:
Screen Shot 2019-11-08 at 4 15 14 PM

Mobile:
Screen Shot 2019-11-08 at 4 15 22 PM

## Why

For #53

## Testing/Questions

Features that this PR affects:

- Pull quotes

Questions that need to be answered before merging:

- [x] Is this PR targeting the correct branch in this repository?
- [ ] WP 5.3 isn't out until next week, _maybe_, which includes the ability to modify alignment variations for blocks. Do we want to just hide these via css for now? https://github.com/INN/umbrella-citylimits/issues/53#issuecomment-544551966
- [ ] The quote icons being next to the borders as detailed in the [design](https://user-images.githubusercontent.com/1754187/67215578-b15e6e80-f3ef-11e9-920b-aaeafd3a0093.png) didn't work because `blockquote` wouldn't let me add psuedo elements to it. Any other ideas or is it ok as is?

Steps to test this PR:

1. Add a pull quote to a post
2. Make sure the color picker works
3. Make sure it looks ok on desktop aligned right, left, default
4. Make sure it looks ok on mobile aligned right, left, default

### CODE REVIEW

**Comment by benlk** [comment]
>> These mockups don't show style variants like Workday did, so we can unregister the WP Core "large" variation https://developer.wordpress.org/block-editor/developers/filters/block-filters/ using the JS methods for WP pre-5.3.
>
> WP 5.3 isn't out until next week, maybe, which includes the ability to modify alignment variations for blocks. Do we want to just hide these via css for now?

I don't think we need to hide the alignment variations. 

The "solid color" variant actually looks pretty good, save for the alignment of the opening `"`:

Screen Shot 2019-11-08 at 17 04 48

If we did want to hide the "Solid color" option, that could be done something like https://developer.wordpress.org/block-editor/developers/filters/block-filters/:

```js
wp.blocks.unregisterBlockStyle( 'core/pullquote', 'solid-color' );
```

I'm not sure what the color settings should do for the pull quote in the default style. Text color succeeds in modifying the text color, but "Main Color" does nothing. Could it change the `"` colors? In the "Solid Color" variant, it sets the background color.

Screen Shot 2019-11-08 at 17 04 04
Screen Shot 2019-11-08 at 17 04 13

> The quote icons being next to the borders as detailed in the design didn't work because blockquote wouldn't let me add psuedo elements to it. Any other ideas or is it ok as is?

I'm fine with it as-is. @MirandaEcho or @kaylima?

File: wp-content/themes/citylimits/inc/block-color-palette.php [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+/**
+ * Define the block color palette
+ *
+ * If updating these colors, please update less/vars.less. Slugs should match LESS var names.
+ *
+ * @link https://developer.wordpress.org/block-editor/developers/themes/theme-support/
+ * @return Array of Arrays
+ */
+function citylimits_block_colors() {
+	return array(
```
Comment (benlk) [inline wp-content/themes/citylimits/inc/block-color-palette.php]: Let's add the Beige CSS variable to this.

**Comment by benlk** [comment]
Alignwide and alignfull have the same appearance, and both go off the left-hand side of the screen by a little bit:

wide:
Screen Shot 2019-11-08 at 17 17 50

full:

Screen Shot 2019-11-08 at 17 18 08

File: wp-content/themes/citylimits/inc/block-color-palette.php [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+			'slug' => 'grey2',
+			'color' => '#e2e2e2',
+		),
+		array(
+			'name' => __( 'Grey 3', 'citylimits' ),
+			'slug' => 'grey3',
+			'color' => '#666',
+		),
+		array(
+			'name' => __( 'Beige', 'citylimits' ),
```
Comment (joshdarby) [inline wp-content/themes/citylimits/inc/block-color-palette.php]: @benlk Regarding https://github.com/INN/umbrella-citylimits/pull/101/files#r344387643, `@beige` is right here

Comment (benlk) [reply wp-content/themes/citylimits/inc/block-color-palette.php]: I wonder what it says about me that I didn't see the color beige in the color list at all. It was just another shade of grey. It's even in half the screenshots I provided.

**Comment by joshdarby** [comment]
> I'm not sure what the color settings should do for the pull quote in the default style. Text color succeeds in modifying the text color, but "Main Color" does nothing. Could it change the " colors? In the "Solid Color" variant, it sets the background color.

@benlk I don't think we can easily change the quote color based on "Main Color" without modifying the block. "Main Color" sets a `border-color` inline style on the `figure` element. We could make the actual borders use whatever is set for "Main Color" though.

**Comment by joshdarby** [comment]
https://github.com/INN/umbrella-citylimits/pull/101/commits/374b74f016f71b5da0bb4d6dd4a1fbc9044405a3 updates the styling so that the "Main Color" option controls the border color.

**Comment by joshdarby** [comment]
> Alignwide and alignfull have the same appearance, and both go off the left-hand side of the screen by a little bit:

@benlk https://github.com/INN/umbrella-citylimits/pull/101/commits/c1a7465fcbeed959f8a9c6a830b22b53af81c50b updates them so `full width` is as wide as possible and `wide width` is just the width of the container

Full width:
Screen Shot 2019-11-11 at 11 51 22 AM

Wide width:
Screen Shot 2019-11-11 at 11 51 13 AM

File: wp-content/themes/citylimits/less/typography.less [hunk 3]
```diff
@@ -37,3 +37,137 @@ a:hover, a:active {
...
+    color: @red;
+    width: 100%;
+    height: 1.5rem;
+    border-top: 1px solid;
+    border-right: 1px solid;
+    border-color: inherit;
+    position: absolute;
+    top: 0;
+    font-size: 60px;
+    margin: -10px;
```
Comment (benlk) [inline wp-content/themes/citylimits/less/typography.less]: with `.is-style-solid-color`, this should be `-10px 10px` to position it inside the background:

Before:
Screen Shot 2019-11-11 at 13 36 03

After:
Screen Shot 2019-11-11 at 13 36 18

**Comment by benlk** [comment]
> "Main Color" sets a border-color inline style on the figure element.

That is a disappointment. ☹️ 
But you're right, there isn't really anything we can do here.

**Comment by joshdarby** [comment]
> with .is-style-solid-color, this should be -10px 10px to position it inside the background:

@benlk https://github.com/INN/umbrella-citylimits/pull/101/commits/6923450867390d880a7ebdf628961889fc64adb1 makes that update.

**Pull request merged**